src/github.com/getlantern/connpool/connpool.go

```diff
@@ -20,17 +20,20 @@ var (
 
 type DialFunc func() (net.Conn, error)
 
-// Pool is a pool of connections.  Connections are pooled eagerly up to MinSize
-// and expire after ClaimTimeout.  Pool attempts to always have MinSize
-// unexpired connections ready to go so that callers don't have to wait on a
-// connection being established when they're ready to use it.
+// Pool is a pool of connections. Connections are pooled lazily up to Size and
+// expire after ClaimTimeout. Lazily here means that the Pool won't start to
+// fill until the first request for a connection. As long as the Pool is
+// actively being used, it will attempt to always have Size number of
+// connections ready to use.
 type Pool struct {
-	// MinSize: the pool will always attempt to maintain at least these many
+	// Size: while active, the pool will attempt to maintain at these many
 	// connections.
-	MinSize int
+	Size int
```

Comment:
As changing `Size` after `Start()` has no impact on how connpool works, we'd better disallow changing at all.

Pass it as a parameter to `Start()`, and rename `actualSize` to `size`, or initialise it through `NewPool()`.